coffeecard/CoffeeCard.WebApi/Controllers/v2/AccountController.cs

```diff
@@ -0,0 +1,52 @@
+using CoffeeCard.Common.Errors;
+using CoffeeCard.Library.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace CoffeeCard.WebApi.Controllers.v2
+{
+    /// <summary>
+    /// Controller for creating and managing user accounts
+    /// </summary>
+    [ApiController]
+    [ApiVersion("2")]
+    [Route("api/v{version:apiVersion}/[controller]")]
+    [Authorize]
+    public class AccountController : ControllerBase
+    {
+        private readonly IAccountService _accountService;
+        
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AccountController"/> class.
+        /// </summary>
+        public AccountController(IAccountService accountService)
+        {
+            _accountService = accountService;
+        }
+
+        /// <summary>
+        /// Delete an account
+        /// </summary>
+        /// <param name="email">email of account to delete</param>
+        /// <response code="204">Successful account deletion</response>
+        /// <response code="404">Account not found</response>
+        /// <response code="401">Invalid credentials</response>
+        [HttpDelete("account")]
```

Comment:
Adding `account` to `HttpDelete` appends the account to the route, which will give the route `/api/v2/Account/account`. I think it would be more RESTful with just `DELETE /api/v2/Account`.

```suggestion
        [HttpDelete]
```